### ISSUE TITLE
Balck shadow artifacts in occluded region

### ISSUE THREAD

**Issue by menzihan** [CUHK-AIM-Group/EndoGaussian#40]
Thanks for sharing your work. I am currently working with the default config from your repository (trained for 3k iterations) and notice severe black artifacts. What modifications would you recommend to remove these artifacts?

<img width="1271" alt="Image" src="https://github.com/user-attachments/assets/5b640561-0b28-4787-ba35-735d3c4d8c0b" />

Additionally, to better align with the published results, would it be possible for you to share the fully trained .pth weights used in your paper? This would greatly assist in verifying our implementation.

**Comment by yifliu3** [comment]
Thank you for your interest in our EndoGaussian work and for bringing this issue to our attention. 

After reviewing your report, I must note that we haven't observed the artifacts you're describing in our implementations. Our rendered outputs typically show proper filling in occluded regions without the black shadows or colored spikes that you mentioned.

It might be helpful to check: 
1. If you're using the most recent version of our code 
2. Your rendering parameters, particularly those related to TV loss
3. Whether you've modified any of the default rendering settings 
4. Whether compile the rasterizer using the same CUDA version

![Image](https://github.com/user-attachments/assets/0e896aca-3a8c-4b3b-8f4a-cbf4ed4bdafe)

![Image](https://github.com/user-attachments/assets/15833c84-1d82-4fe1-83f5-471979af142f)

**Comment by zhmen3** [comment]
Thank you for your quick response, Liu. During the adaptation process, I do made some modifications to the rendering part of the previous code. 

Based on your suggestions, I pulled the latest code again and redeployed the virtual environment (py3.7 cuda 11.7) as instructed. Using the default code and config files, the results for 3k iterations are as follows:
<img width="1083" alt="Image" src="https://github.com/user-attachments/assets/ace43729-1e32-4d6b-a849-3288f88e8020" />

The rendering results for 20k iterations using the default xx_MLP.py are shown below (with PSNR~5 for the pulling scene at this stage):
<img width="1079" alt="Image" src="https://github.com/user-attachments/assets/a6b867cd-3d73-48b7-b8fc-c6e6cc74047c" />

<img width="589" alt="Image" src="https://github.com/user-attachments/assets/fece2c3b-1142-4d32-b1e1-a60634d9356f" />

Anything missed? or if possible, could you share the pre-trained model:?

**Comment by yifliu3** [comment]
Thanks for your attention. I just realized that your problem is similar to the one raised at https://github.com/CUHK-AIM-Group/EndoGaussian/issues/11 . Maybe you can check that first.

For easy to reproduce the results, we will release the pre-trained model later today.

**Comment by yifliu3** [comment]
The link for downloading checkpoints has been updated. 
https://mycuhk-my.sharepoint.com/:f:/g/personal/1155195605_link_cuhk_edu_hk/EtfmmU8HsaRCnpZOUqoF_kAB8790U28aRXSQ-ngHPhd4qg

**Comment by yifliu3** [comment]
This is strange, and maybe you can first check if the gaussian initialization code has been modified. From the image the number of points after initializing is 38125, while we have set to sample fixed 30000 points.

**Comment by zhmen3** [comment]
I see. In your README file, you just refer to another project's repo. When I tried to render & eval your pretrained model on the exact project, the visual output appeared normal finally;)
 
<img width="1131" alt="Image" src="https://github.com/user-attachments/assets/93562ce3-92af-4d5e-a465-dce917c7fa18" />

However, I'm unsure why the metrics printed are abnormally low. May you check that?
<img width="802" alt="Image" src="https://github.com/user-attachments/assets/3a1e49e3-4c65-48db-8d7c-c115f4711c51" />
<img width="803" alt="Image" src="https://github.com/user-attachments/assets/9c0aca98-aecb-4663-b706-9cad9c72c9a0" />

**Comment by zhmen3** [comment]
I got very similar eval result even using other matrics implementation. May you check that?

<img width="568" alt="Image" src="https://github.com/user-attachments/assets/bc31c91f-d353-43ea-82b4-0c7390f086dd" />

**Comment by yifliu3** [comment]
That's strange and we haven't encountered such problems. Here are some potential reasons:
1. The code has been modified
2. The data is broken
3. The paths of data is not correct
